include/alpaka/mem/buf/BufOmp5.hpp

```diff
@@ -0,0 +1,713 @@
+                        }
+
+                    public:
+                        //-----------------------------------------------------------------------------
+                        //! Constructor
+                        template<
+                            typename TExtent>
+                        ALPAKA_FN_HOST BufOmp5Impl(
+                            dev::DevOmp5 const & dev,
+                            TElem * const pMem,
+                            TExtent const & extent) :
+                                m_dev(dev),
+                                m_extentElements(extent::getExtentVecEnd<TDim>(extent)),
+                                m_pitchBytes(calculatePitchesFromExtents(m_extentElements)),
+                                m_pMem(pMem)
+                        {
+                            ALPAKA_DEBUG_MINIMAL_LOG_SCOPE;
+
+                            static_assert(
+                                TDim::value == dim::Dim<TExtent>::value,
+                                "The dimensionality of TExtent and the dimensionality of the TDim template parameter have to be identical!");
+                            static_assert(
+                                std::is_same<TIdx, idx::Idx<TExtent>>::value,
+                                "The idx type of TExtent and the TIdx template parameter have to be identical!");
+                        }
+
+                    public:
+                        dev::DevOmp5 m_dev;
+                        vec::Vec<TDim, TIdx> m_extentElements;
+                        vec::Vec<TDim, TIdx> m_pitchBytes;
+                        TElem* m_pMem;
+
+                            BufOmp5Impl(const BufOmp5Impl&) = delete;
+                            BufOmp5Impl(BufOmp5Impl &&) = default;
+                        BufOmp5Impl& operator=(const BufOmp5Impl&) = delete;
+                        BufOmp5Impl& operator=(BufOmp5Impl&&) = default;
+                            ~BufOmp5Impl()
+                        {
+                            omp_target_free(m_pMem, m_dev.m_spDevOmp5Impl->iDevice());
+                        }
+                    };
+                }
+            }
+            template<
+                typename TElem,
+                typename TDim,
+                typename TIdx>
+            class BufOmp5
+            {
+            public:
+                //-----------------------------------------------------------------------------
+                //! Constructor
+                template<
+                    typename TExtent>
+                ALPAKA_FN_HOST BufOmp5(
+                    dev::DevOmp5 const & dev,
+                    TElem * const pMem,
+                    TExtent const & extent) :
+                        m_spBufImpl(std::make_shared<omp5::detail::BufOmp5Impl<TElem, TDim, TIdx>>(dev, pMem, extent))
+                {}
+
+                    BufOmp5(const BufOmp5&) = default;
+                    BufOmp5(BufOmp5 &&) = default;
```

Comment:
indentation seems to be wrong

Reply:
Habit (indenting ctors and dtors), sorry. Fixed.